firmware/application/external/rook/ui_rook.cpp

```diff
@@ -0,0 +1,269 @@
+#include "portapack.hpp"
+#include "io_file.hpp"
+#include "ui_fileman.hpp"
+#include "file_path.hpp"
+#include <cstring>
+#include "baseband_api.hpp"
+#include "ui_textentry.hpp"
+#include "encoders.hpp"         // Includes data encoding functions for transmission
+#include "baseband_api.hpp"     // Includes baseband API for handling transmission settings
+#include "file_reader.hpp"
+ 
+#define PADDING_LEFT 1
+#define PADDING_RIGHT 1
+ 
+using namespace portapack;
+using namespace ui;
+namespace fs = std::filesystem;
+ 
+namespace ui::external_app::rook {
+ 
+    void ROOKAppView::focus() {
+        button_set.focus();
+    }
+ 
+    // `start_tx` method: Configures and begins OOK data transmission with a specific message.
+    void ROOKAppView::start_tx(const std::string& message) {
+        size_t bitstream_length = encoders::make_bitstream(const_cast<std::string&>(message));  // Convert the message into a bitstream
+        // Retrieve selected sample rate using selected_index_value()
+        int32_t SAMPLE_RATE_VALUE = field_sample_rate.selected_index_value();
+        int32_t SYMBOL_RATE_VALUE = cant_symbol_rate.value();
+        int32_t REPEAT = cant_repeat.value();
+        int32_t PAUSE_SYMBOL = cant_pause_symbol.value();
+ 
+        transmitter_model.set_sampling_rate(SAMPLE_RATE_VALUE);         // Set the OOK sampling rate
+        transmitter_model.set_baseband_bandwidth(1750000);              // Set the baseband bandwidth
+        transmitter_model.enable();                                     // Enable the transmitter
+ 
+        // Configure OOK data and transmission characteristics
+        baseband::set_ook_data(
+            bitstream_length,                 // Length of the bitstream to transmit
+            SAMPLE_RATE_VALUE / SYMBOL_RATE_VALUE, // Calculate symbol period based on repetition rate
+            REPEAT,                           // Set the number of repetitions per symbol
+            PAUSE_SYMBOL                      // Set the pause between symbols
+        );
+    }
+ 
+    // `stop_tx` method: Stops the transmission and resets the progress bar.
+    void ROOKAppView::stop_tx() {
+        transmitter_model.disable();                   // Disable the transmitter
+        progressBar_progress.set_value(0);             // Reset progress bar to 0
+    }
+ 
+    // `on_file_changed` method: Called when a new file is loaded; parses file data into variables
+    void ROOKAppView::on_file_changed(const fs::path& new_file_path) {
+        payload.clear();  // Clear previous payload content
+ 
+        File data_file;
+        auto open_result = data_file.open(new_file_path);
+        if (open_result) {
+            nav_.display_modal("Error", "Unable to open file.");
+            return;
+        }
+ 
+        FileLineReader reader(data_file);
+        for (const auto& line : reader) {
+            // Split the line into segments to extract each value
+            size_t first_space = line.find(' ');
```

Comment:
i would probably use `,` instead, so it can match the common designing standard.

But also I think I understand your point too, maybe in some country the `,` would be considered as decimal identifier and cause misleading, so you use space?

i think either is fine to me but FYI the playlist file use `,`

These are just suggestions. you are free to choose accept or not

Reply:
For now I decided to keep it like this, if anything I make that change in the future, hopefully it won't be a problem.